### PR TITLE
[DQT] Fix react errors 

### PR DESCRIPTION
**Pull request by zaliqarosli**
### Brief summary of changes

- removes console log errors
- includes changes from https://github.com/aces/Loris/pull/4464 which i will now close
- replaces functions with arrow functions for automatic binding, allowing us to remove silly things like `var that = this`

### This resolves issue...

- [ ] Github? # https://github.com/aces/Loris/issues/4731

### To test this change...

- [ ] Go onto DQT on this branch and make sure there are absolutely no more errors in your front-end console log

### CODE REVIEW

File: modules/dataquery/jsx/react.filterBuilder.js [hunk 1]
```diff
@@ -78,13 +78,12 @@ class FilterRule extends Component {
 
   selectInstrument(event) {
     // Update the rules instrument, getting the instruments avalible fields
-    let rule = this.props.rule,
-      that = this;
+    let rule = this.props.rule;
     if (event.target.value) {
       rule.instrument = event.target.value;
-      $.get(loris.BaseURL + '/AjaxHelper.php?Module=dataquery&script=datadictionary.php', {category: rule.instrument}, function(data) {
+      $.get(loris.BaseURL + '/AjaxHelper.php?Module=dataquery&script=datadictionary.php', {category: rule.instrument}, (data) => {
```
Comment (driusan) [inline modules/dataquery/jsx/react.filterBuilder.js]: I think this URL should be /dataquery/ajax/datadictionary.php

**Comment by driusan** [comment]
@sruthymathew123 do you have somewhere you can manually test this?

**Comment by sruthymathew123** [comment]
@driusan LGTM
dqt_sucesss

**Comment by zaliqarosli** [comment]
@driusan done!

**Comment by driusan** [comment]
I noticed a few more when I did ctrl-f to verify it was fixed, but I'll leave those for now instead of risking breaking anything by changing that many URLs right before the release

**Pull request merged**